internal/ir/xds.go

```diff
@@ -224,8 +224,20 @@ type CoreListenerDetails struct {
 	ExtensionRefs []*UnstructuredRef `json:"extensionRefs,omitempty" yaml:"extensionRefs,omitempty"`
 	// Metadata is used to enrich envoy resource metadata with user and provider-specific information
 	Metadata *ResourceMetadata `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+	// IPFamily specifies the IP address family for the gateway.
+	// It can be IPv4, IPv6, or Dual.
+	IPFamily IPFamily `json:"ipFamily,omitempty" yaml:"ipFamily,omitempty"`
```

Comment:
can you make it a pointer? so that this PR won't be that huge.

Reply:
> I'll post it separately as another PR.

I changed here.